src/lavinmq/amqp/queue/queue.cr

```diff
@@ -190,13 +190,39 @@ module LavinMQ::AMQP
         @paused.set(true)
       end
       @msg_store = init_msg_store(@data_dir)
+      @empty = @msg_store.empty
+      start
+    end
+
+    private def start
       if @msg_store.closed
         close
+      else
+        handle_arguments
+        spawn queue_expire_loop, name: "Queue#queue_expire_loop #{@vhost.name}/#{@name}" if @expires
+        spawn message_expire_loop, name: "Queue#message_expire_loop #{@vhost.name}/#{@name}"
       end
+    end
+
+    def restart!
+      return unless @closed
+      reset_queue_state
+      @msg_store = init_msg_store(@data_dir)
       @empty = @msg_store.empty
-      handle_arguments
-      spawn queue_expire_loop, name: "Queue#queue_expire_loop #{@vhost.name}/#{@name}" if @expires
-      spawn message_expire_loop, name: "Queue#message_expire_loop #{@vhost.name}/#{@name}"
+      start
+    end
+
+    private def reset_queue_state
+      @closed = false
+      @state = QueueState::Running
+      # Recreate channels that were closed
+      @queue_expiration_ttl_change = ::Channel(Nil).new
+      @message_ttl_change = ::Channel(Nil).new
+      @paused = BoolChannel.new(false)
```

Comment:
Should we check if the queue was paused and if so reset into pasued state?

Reply:
yeah, we can probably just move the paused state handling to start